w3-replication.md

```diff
@@ -0,0 +1,397 @@
+
+##### Space
+
+The `nb.space` field MUST be set to the (byte encoded) [DID] of the user space where allocation took place.
+
+##### Location
+
+The `nb.site` field MUST be a [Link] to the [location commitment] describing where the blob can be retrieved.
+
+##### Cause
+
+The `nb.cause` field MUST be set to the [Link] for the [Replicate Blob](#blob-replicate) task, that caused an allocation.
+
+#### Blob Replica Allocate Receipt Schema
+
+```ts
+type AllocateReplicaBlobReceipt = {
+  ran:  Link<AllocateReplicaBlob>
+  out:  Result<AllocateReplicaBlobOk, AllocateReplicaBlobError>
+  fx: {
+    fork: [Link<TransferReplicaBlob>]
+  }
+}
+
+type AllocateReplicaBlobOk = {
+  /** The number of bytes allocated for a Blob. */
+  size: int
+}
+
+type AllocateReplicaBlobError = {
+  message: string
+}
+```
+
+##### Blob Replica Allocate Result
+
+Invocation MUST fail if any of the following is true:
+
+1. Provided `blob.size` is outside of supported range.
+1. Provided `blob.digest` is not a valid [multihash].
+1. Provided `blob.digest` [multihash] hashing algorithm is not supported.
+1. Provided `site` is a valid, non-revoked [location commitment].
+
+Invocation MUST succeed if non of the above is true.
+
+###### Size
+
+The `out.ok.size` MUST be set to the number of bytes that were allocated for the `Blob`. It MUST be equal to either:
+
+1. The `nb.blob.size` of the invocation.
+2. `0` if the storage node already has memory allocated for the `nb.blob`.
+
+##### Blob Replica Allocate Effects
+
+Successful invocation MUST start a workflow consisting of a [Transfer Replica Blob](#blob-replica-transfer) task, that MUST be set in receipt effects (`fx` field).
+
+### Blob Replica Transfer
+
+The `blob/replica/transfer` task is completed by each replication node. Replication data is fetched from the location specified in the location commitment referenced by the `blob/replica/allocate` invocation.
+
+A `blob/replica/transfer` task takes the following form:
+
+```json5
+{
+  "iss": "did:key:zStorageNode",
+  "aud": "did:key:zStorageNode",
+  "att": [
+    {
+      "with": "did:key:zStorageNode",
+      "can": "blob/replica/transfer",
+      "nb": {
+        /** The blob that will be transferred. */
+        "blob": {
+          "digest": { "/": { "bytes": "..." } },
+          "size": 1234
+        },
+        /** DID of the space the blob has been allocated to. */
+        "space": { "/": { "bytes": "..." } },
+        /** The location the blob will be transferred from. */
+        "site": { "/": "bafy..locationCommitment" },
+        /** The `blob/replica/allocate` invocation that initiated this transfer. */
+        "cause": { "/": "bafy..allocate" }
+      }
+    }
+  ],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+When the `blob/replica/transfer` task is complete a receipt is issued. The receipt is communicated back to the upload service via a [`ucan/conclude` invocation](./w3-ucan.md#conclusion).
```

Comment:
Is it worth discussing failure cases here for transfer? Or is it intended that the replica node performs the transfer task until a successful completion? Is it the responsibility of the upload service to identify failed transfers and re-schedule invocations to complete them?